### PR TITLE
Add Util module

### PR DESCRIPTION
**Pull request by lavoiesl**
Various utility methods, including some from ActiveSupport

- `camelCase` to `snake_case` and `dash-case` conversion.
- Strip Heredoc
- english join (with oxford comma)
- Various block executors like `with_environment` and `with_tmp_dir`
- Filesize and SI scale conversion
- Retrier

### CODE REVIEW

File: lib/cli/kit/util.rb [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            .gsub(/([A-Z]+)([A-Z][a-z])/, "\\1#{seperator}\\2") # MyCoolThing::MyAPI_Module
+            .gsub(/([a-z\d])([A-Z])/, "\\1#{seperator}\\2") # My_Cool_Thing::My_API_Module
+            .downcase # my_cool_thing/my_api_module
+        end
+
+        def dash_case(camel_case)
+          snake_case(camel_case, '-')
+        end
+
+        # The following methods is taken from activesupport
```
Comment (jules2689) [inline lib/cli/kit/util.rb]: I think we need to copy the MIT license in full here

Comment (lavoiesl) [reply lib/cli/kit/util.rb]: Considering even the rails project only includes a link to the license, I’ll do the same here.

File: lib/cli/kit/util.rb [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        # LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+        # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+        # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+        #
+        # Strips indentation by removing the amount of leading whitespace in the least indented
+        # non-empty line in the whole string
+        #
+        def strip_heredoc(str)
+          str.gsub(/^#{str.scan(/^[ \t]*(?=\S)/).min}/, "".freeze)
+        end
```
Comment (thegedge) [inline lib/cli/kit/util.rb]: With the preference of `<<~` over `<<-`, we probably won't need this method (if you wanted to have less things floating around in here).

File: lib/cli/kit/util.rb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+          small_scale = %w(m µ n p f a z y)
+          big_scale = %w(k M G T P E Z Y)
+          negative = number < 0
+          number = number.abs.to_f
+
+          if number == 0 || number.between?(1, factor)
+            prefix = ""
+            scale = 0
+          else
+            scale = Math.log(number, factor).floor
```
Comment (thegedge) [inline lib/cli/kit/util.rb]: We can follow up in another PR, but I would highly recommend rewriting this to not use `Math.log`:

```
irb(main):004:0> Math.log(1000, 10).floor
=> 2
```

😭

File: lib/cli/kit/util.rb (hunk 2 above)
Comment (lavoiesl) [reply lib/cli/kit/util.rb]: It's still used a bit everywhere, so I’d prefer keeping it.

File: lib/cli/kit/util.rb (hunk 3 above)
Comment (lavoiesl) [reply lib/cli/kit/util.rb]: I actually rewrote this function completely during this PR. Why not use log? It seems more proper than a loop dividing the number over and over.

Comment (lavoiesl) [reply lib/cli/kit/util.rb]: Factor of 10 is not expected, it's only 1000 or 1024. I guess I could raise if that's the case.

Comment (thegedge) [reply lib/cli/kit/util.rb]: As long as both factors are correct for all of the prefixes we have (when the result we'd expect should be exactly `1 prefix+unit`), that can work too 👍

Comment (lavoiesl) [reply lib/cli/kit/util.rb]: Done.

**Pull request merged**